tests/common/devices/sonic.py

```diff
@@ -642,31 +637,47 @@ def all_critical_process_status(self):
             service_group_process = group_process_results[service]
 
             service_result = service_results[service]
-            # If container is not running, stdout_lines is empty
+
+            all_critical_process[service] = self.get_service_status_and_critical_process(
+                service_result=service_result,
+                service_critical_process=service_critical_process,
+                critical_group_list=service_group_process['groups'],
+                critical_process_list=service_group_process['processes']
+            )
+
+        return all_critical_process
+
+    def get_service_status_and_critical_process(self, service_result, service_critical_process, critical_group_list,
```

Comment:
This function is to do parsing. Suggest to rename it to something like `parse_xxx`.